src/server/game/Entities/Creature/Creature.h

```diff
@@ -101,6 +101,14 @@ class TC_GAME_API Creature : public Unit, public GridObject<Creature>, public Ma
         bool HasReactState(ReactStates state) const { return (m_reactState == state); }
         void InitializeReactState();
 
+        // These shadow the definitions on Unit and override the default value for keepCombat
```

Comment:
mhm

i'm also wondering if we ever need the method on base unit. is there ever a situation where non-creatures are set as immune_to_x?